clientapi/routing/routing.go

```diff
@@ -129,6 +129,10 @@ func Setup(
 		})).Methods(http.MethodGet, http.MethodOptions)
 	}
 
+	if !strings.HasPrefix(string(cfg.Matrix.ServerName), "http://") || !strings.HasPrefix(string(cfg.Matrix.ServerName), "https://") {
+		logrus.Warn("The well_known_server_name does not start with http:// or https:// does not start with 'http://' or 'https://'. Some clients may fail to connect.")
```

Comment:
Looks like accidentally duplication? (as well as `well_known_client_name` instead of `well_known_server_name`)
Also better to put this inside the `if cfg.Matrix.WellKnownClientName != "" {` block above. (same for the federation change below)